README.adoc: add common rationales for Swarm in the intro (#318)
jenkinsci/swarm-plugin: README.adoc

== Introduction

The Swarm plugin enables nodes to join a nearby Jenkins controller, thereby forming an ad-hoc cluster.

> This plugin makes it easier to scale a Jenkins cluster by spinning up and tearing down new nodes, for team members to contribute compute resources to a build farm, or to attach builders that the Jenkins controller can not initiate connections to.

Reviewer: ```suggestion
This plugin makes it easier to scale a Jenkins cluster by spinning up and tearing down new agents, enabling team members to contribute compute resources to a build farm or to attach agents to which the Jenkins controller cannot initiate connections.
```

== Getting started

. Install the Swarm plugin from the Update Center.

> . Download https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/swarm-client/[the Swarm client] on your worker machine or container (JRE 8+ also needed there).

Reviewer: ```suggestion
. Ensure your agent is running version 8 or later of the Java Runtime Environment (JRE). The recommendation is to use the same JRE distribution and version as the controller.
. Download https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/swarm-client/[the Swarm client] on your agent.
```

> . Run the Swarm client with `java -jar path/to/swarm-client.jar -master JENKINS_URL -username SWARM_ACCOUNT -password...` (note there are several ways to pass a password/token, of varying security; the account should have permissions set up to interact with the swarm-plugin on server side -- see Security document referenced below for details). There are no other required command-line options; run with the `-help` option to see the available options.

Reviewer: ```suggestion
. Run the Swarm client with `java -jar path/to/swarm-client.jar -master ${JENKINS_URL} -username ${USERNAME}` and one of the authentication options as described in xref:docs/security.adoc#authentication[the Global Security Configuration documentation]. There are no other required command-line options; run with the `-help` option to see the available options.
```
